### PR TITLE
Update Readme.md

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,23 +1,81 @@
...
 <!-- default badges end -->
-<!-- default file list -->
-*Files to look at*:
 
-* [HomeController.cs](./CS/DXWebApplication1/Controllers/HomeController.cs) (VB: [HomeController.vb](./VB/DXWebApplication1/Controllers/HomeController.vb))
-* **[GridViewPartial.cshtml](./CS/DXWebApplication1/Views/Home/GridViewPartial.cshtml)**
-* [Index.cshtml](./CS/DXWebApplication1/Views/Home/Index.cshtml)
-<!-- default file list end -->
-# GridView - How to track ClientLayout with a separate ListBox
+# Grid View for ASP.NET MVC - How to use a list box editor to save and restore client layout
```
Comment (arminal) [inline Readme.md]: I understand why you wrote "restore client layout", but I think in this case, it is better to say " How to use a list box editor to save and restore end-user layouts"

File: Readme.md [hunk 2]
```diff
@@ -1,23 +1,81 @@
...
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/t146962/)**
 <!-- run online end -->
 
+This example demonstrates how to save and restore client layouts of the grid. The modified layouts are stored in a [ListBox](https://docs.devexpress.com/AspNetMvc/8990/components/data-editors-extensions/listbox) editor.
```
Comment (arminal) [inline Readme.md]:
```suggestion
This example demonstrates how to save and restore grid layouts created by end-users. These modified layouts are stored in a [ListBox](https://docs.devexpress.com/AspNetMvc/8990/components/data-editors-extensions/listbox) editor.
```

File: Readme.md [hunk 3]
```diff
@@ -1,23 +1,81 @@
...
+
+![](grid-client-layout-in-listbox.png)
+
+## Implementation Details
+
+1. Handle the [GridViewSettings.CustomJSProperties](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.GridViewSettings.CustomJSProperties) event and call the [MVCxGridView.SaveClientLayout](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridBase.SaveClientLayout) method to save the currently applied client layout.
```
Comment (arminal) [inline Readme.md]:
```suggestion
1. Handle the [GridViewSettings.CustomJSProperties](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.GridViewSettings.CustomJSProperties) event and call the [MVCxGridView.SaveClientLayout](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridBase.SaveClientLayout) method to save the currently applied layout.
```

File: Readme.md [hunk 4]
```diff
@@ -1,23 +1,81 @@
...
+
+  ```csharp
+  settings.CustomJSProperties = (s, e) => {
+      MVCxGridView gridView = (MVCxGridView)s;
+      e.Properties["cpClientLayout"] = gridView.SaveClientLayout();
+  };
+  ```
+
+2. Handle [ASPxClientGridView.Init](https://docs.devexpress.com/AspNet/js-ASPxClientControlBase.Init) and [ASPxClientGridView.EndCallback](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.EndCallback) events to store the currently applied client layout in ListBox.
```
Comment (arminal) [inline Readme.md]:
```suggestion
2. Handle [ASPxClientGridView.Init](https://docs.devexpress.com/AspNet/js-ASPxClientControlBase.Init) and [ASPxClientGridView.EndCallback](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.EndCallback) events to store the currently applied layout in ListBox.
```

File: Readme.md [hunk 5]
```diff
@@ -1,23 +1,81 @@
...
+      TrackLayout(s.cpClientLayout);
+  }
+  function TrackLayout(layout) {
+      if (!lb.FindItemByValue(layout))
+          lb.AddItem(layout);
+      lb.SetValue(layout);
+  }
+  ```
+   
+3. Handle the [ASPxClientListBox.SelectedIndexChanged](https://docs.devexpress.com/AspNet/js-ASPxClientListBox.SelectedIndexChanged) event to perform a custom grid callback (call the [MVCxClientGridView.PerformCallback(data)](https://docs.devexpress.com/AspNetMvc/js-MVCxClientGridView.PerformCallback(data)) method) when the selected item changes. Pass the selected client layout to the method as a parameter.
```
Comment (arminal) [inline Readme.md]:
```suggestion
3. Handle the [ASPxClientListBox.SelectedIndexChanged](https://docs.devexpress.com/AspNet/js-ASPxClientListBox.SelectedIndexChanged) event to perform a custom grid callback (call the [MVCxClientGridView.PerformCallback(data)](https://docs.devexpress.com/AspNetMvc/js-MVCxClientGridView.PerformCallback(data)) method) when the selected item changes. Pass the end-user's selected layout to the method as a parameter.
```

File: Readme.md [hunk 6]
```diff
@@ -1,23 +1,81 @@
...
 
-<br/>
+  ```jscript
+  function OnSelectedIndexChanged(s, e) {
+      gv.PerformCallback({ layoutToApply : s.GetValue() });
+  }
+  ```
+
+4. Handle the custom grid callback in a separate Action method. Retrieve the client layout to be applied, and save this value in `ViewData`.
```
Comment (arminal) [inline Readme.md]:
```suggestion
4. Handle the custom grid callback in a separate Action method. Retrieve the  layout the end-user wants to appy and save this value in `ViewData`.
```

File: Readme.md [hunk 7]
```diff
@@ -1,23 +1,81 @@
...
+
+## Files to Review
+
+* [HomeController.cs](./CS/DXWebApplication1/Controllers/HomeController.cs) (VB: [HomeController.vb](./VB/DXWebApplication1/Controllers/HomeController.vb))
+* [GridViewPartial.cshtml](./CS/DXWebApplication1/Views/Home/GridViewPartial.cshtml)
+* [Index.cshtml](./CS/DXWebApplication1/Views/Home/Index.cshtml)
 
+## More Examples
 
+* [Grid View for ASP.NET Web Forms - How to use a list box editor to save and restore client layout](https://github.com/DevExpress-Examples/asp-net-web-forms-grid-use-listbox-to-save-and-restore-client-layout)
```
Comment (arminal) [inline Readme.md]:
```suggestion
* [Grid View for ASP.NET Web Forms - How to use a list box editor to save and restore grid layouts created by end-users](https://github.com/DevExpress-Examples/asp-net-web-forms-grid-use-listbox-to-save-and-restore-client-layout)
```

File: Readme.md [hunk 8]
```diff
@@ -1,23 +1,81 @@
...
+* [Grid View for ASP.NET MVC - How to save and restore client layout within a session](https://github.com/DevExpress-Examples/asp-net-mvc-grid-save-restore-client-layout-within-a-session)
```
Comment (arminal) [inline Readme.md]:
```suggestion
* [Grid View for ASP.NET MVC - How to save and restore layouts created by end-users within a session](https://github.com/DevExpress-Examples/asp-net-mvc-grid-save-restore-client-layout-within-a-session)
```

File: Readme.md (hunk 1 above)
Comment (LanaDX) [reply Readme.md]:
```suggestion
# Grid View for ASP.NET MVC - How to use a list box editor to save and restore end-user layouts
```

**Pull request merged**